src/cdk/menu/menu-trigger.ts

```diff
@@ -130,7 +129,8 @@ export class CdkMenuTrigger extends CdkMenuTriggerBase implements OnDestroy {
     switch (event.keyCode) {
       case SPACE:
       case ENTER:
-        if (!hasModifierKey(event)) {
+        // Skip events that will trigger clicks so the handler doesn't get triggered twice.
+        if (!hasModifierKey(event) && !eventDispatchesNativeClick(this._elementRef, event)) {
```

Comment:
OOC: Why are we not always just handling the `click` event. Are we doing this for the focus origin?

Reply:
The problem is that the menu trigger does the same both enter/space events and click events. When the enter event trigger a click, it ends up closing and reopening the menu immediately. Previously we were handling it by calling `preventDefault`, but that ended up breaking links used in the menu.

Reply:
I see. I don't quite follow why we handle `SPACE/ENTER` in the first place via `keydown`, but I'm sure there are some nuances I'm just missing here. In either case the change LGTM 😄

Reply:
Ah, we need to do it because click events only get dispatched on space/enter for buttons and links, but the menu trigger and menu item can be set on any arbitrary element.

Reply:
Gotcha